### PR TITLE
imfile: add error message when file not found and parameter to disable the message

### PR DESCRIPTION
**Pull request by PascalWithopf**
As default imfile returns an error if file is not found, but if the parameter is set off then no error is returned.
Closes https://github.com/rsyslog/rsyslog/issues/1535

### CODE REVIEW

File: tests/imfile-file-not-found-error.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+input(type="imfile" File="testsuites/NotExistingInputFile" Tag="tag1")
+
+action(type="omfile" file="rsyslog.out.log")
+'
+. $srcdir/diag.sh startup
+. $srcdir/diag.sh shutdown-when-empty
+. $srcdir/diag.sh wait-shutdown
+
+grep "imfile: error.*No such file or directory" rsyslog.out.log > /dev/null
```
Comment (rgerhards) [inline tests/imfile-file-not-found-error.sh]: The file name should also be checked in the test. I think it is mandatory that it is given, else the user gets confused.

File: tests/imfile-fileNotFoundError-parameter.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+input(type="imfile" File="testsuites/NotExistingInputFile" Tag="tag1" fileNotFoundError="off")
+
+action(type="omfile" file="rsyslog.out.log")
+'
+. $srcdir/diag.sh startup
+. $srcdir/diag.sh shutdown-when-empty
+. $srcdir/diag.sh wait-shutdown
+
+grep "imfile: error.*No such file or directory" rsyslog.out.log > /dev/null
+if [ $? -eq 0 ]; then
```
Comment (rgerhards) [inline tests/imfile-fileNotFoundError-parameter.sh:17]: mhh... isn't it an error if the error message *is* emitted in this case?

Comment (ghost) [reply tests/imfile-fileNotFoundError-parameter.sh:17]: yes, that's why I'm testing if the result of grep search equals 0

Comment (rgerhards) [reply tests/imfile-fileNotFoundError-parameter.sh:17]: lol, you got me ;-)

**Pull request merged**